include/LightGBM/cuda/cuda_split_info.hpp

```diff
@@ -0,0 +1,108 @@
+/*!
+ * Copyright (c) 2021 Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE file in the project root for
+ * license information.
+ */
+
+#ifdef USE_CUDA_EXP
+
+#ifndef LIGHTGBM_TREELEARNER_CUDA_CUDA_SPLIT_INFO_HPP_
+#define LIGHTGBM_TREELEARNER_CUDA_CUDA_SPLIT_INFO_HPP_
+
+#include <LightGBM/meta.h>
+
+namespace LightGBM {
+
+class CUDASplitInfo {
```

Comment:
all these cuda object classes (this and above and after), can we reuse them? at least reduce the duplicated codes.

Reply:
These classes are redefined to avoid some modifications in the original CPU classes. I will combine the CPU and CUDA classes together in subsequent PRs.